commit 2542322485be45853cc72d542d8ed84fae82c981
Author: Steve Wise <swise@opengridcomputing.com>
Date:   Thu Feb 11 15:40:29 2010 -0800

    RDMA/cxgb3: Remove BUG_ON() on CQ rearm failure
    
    Failure to rearm a CQ means the cxgb3 device is wedged, but we shouldn't
    kill the whole system with a BUG_ON() if this happens.
    
    Signed-off-by: Steve Wise <swise@opengridcomputing.com>
    Signed-off-by: Roland Dreier <rolandd@cisco.com>

diff --git a/drivers/infiniband/hw/cxgb3/cxio_hal.c b/drivers/infiniband/hw/cxgb3/cxio_hal.c
index 0677fc7..8837d56 100644
--- a/drivers/infiniband/hw/cxgb3/cxio_hal.c
+++ b/drivers/infiniband/hw/cxgb3/cxio_hal.c
@@ -109,7 +109,6 @@ int cxio_hal_cq_op(struct cxio_rdev *rdev_p, struct t3_cq *cq,
 		while (!CQ_VLD_ENTRY(rptr, cq->size_log2, cqe)) {
 			udelay(1);
 			if (i++ > 1000000) {
-				BUG_ON(1);
 				printk(KERN_ERR "%s: stalled rnic\n",
 				       rdev_p->dev_name);
 				return -EIO;
